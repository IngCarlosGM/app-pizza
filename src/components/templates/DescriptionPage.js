import React from 'react'
import PropTypes from 'prop-types'

export const DescriptionPage = ({ description, myStyle= "" }) => {
    return (
        <div className={`description-page ${ myStyle }`}>
            { description }
        </div>
    )
}

DescriptionPage.propTypes = {
    description: PropTypes.string.isRequired,
    myStyle: PropTypes.string
}
