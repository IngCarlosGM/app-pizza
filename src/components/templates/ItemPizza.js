import React from 'react';
import PropTypes from 'prop-types';

export const ItemPizza = ({ itemPizza }) => {

    return (
        <div className="container-item-store pointer animate__bounceIn">
            <div className="item-store">
                <img 
                    className="img-item-store"
                    src={ `../assets/pizzas/pizza-${itemPizza.id}.png` } 
                    alt="Logo pizza" 
                />
            </div>
            <div className="flex-column">
                <span className="title-item-store">{ itemPizza.name }</span>
            </div>
        </div>
    )
}

ItemPizza.propTypes = {
    itemPizza: PropTypes.object.isRequired
}