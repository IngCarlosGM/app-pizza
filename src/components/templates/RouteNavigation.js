import React from 'react';
import PropTypes from 'prop-types';

export const RouteNavigation = ({ route }) => {
    return (
        <span className="route-navigation">
            { route }
        </span>
    )
}

RouteNavigation.propTypes = {
    route: PropTypes.string.isRequired
}