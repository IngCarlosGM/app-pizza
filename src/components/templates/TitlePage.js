import React from 'react'
import PropTypes from 'prop-types'

export const TitlePage = ({ title, myStyle= "" }) => {
    return (
        <h1 className={` ${ myStyle }`}>
            { title }
        </h1>
    )
}

TitlePage.propTypes = {
    title: PropTypes.string.isRequired,
    myStyle: PropTypes.string
}