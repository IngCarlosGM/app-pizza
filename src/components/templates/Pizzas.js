import React from 'react';

import { ItemPizza } from './ItemPizza';

export const Pizzas = ({ pizzas }) => {

    return (
        <div className="grid-pizzas mt-third">
            {
                pizzas && pizzas.map( itemPizza => {
                    return <ItemPizza 
                                key={ `itemPizza${ itemPizza.id }` } 
                                itemPizza={ itemPizza }
                            />
                })
            }
        </div>
    )
}