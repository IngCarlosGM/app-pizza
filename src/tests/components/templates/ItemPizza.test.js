import { shallow } from "enzyme";
import { ItemPizza } from "../../../components/templates/ItemPizza";
import dataApp from '../../data/dataApp.json';

describe('Testing ItemPizza component', () => {

    const itemPizza = dataApp.response.stores[0].products[0];

    const wrapper = shallow(
        <ItemPizza 
            itemPizza= { itemPizza }
        />
    );
   
    test('should render correctly', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should to have a image correct', () => {
        expect( wrapper.find('img').at(0).prop('src') ).toBe( `../assets/pizzas/pizza-${ itemPizza.id }.png` );
    });

    test('should to have a image correct', () => {
        expect( wrapper.find('span').at(0).text() ).toBe( itemPizza.name );
    });
    
});
