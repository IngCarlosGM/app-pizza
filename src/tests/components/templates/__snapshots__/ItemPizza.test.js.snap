// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing ItemPizza component should render correctly 1`] = `
<div
  className="container-item-store container-item-store--without-hover pointer animate__bounceIn"
>
  <div
    className="item-store"
  >
    <img
      alt="Logo pizza"
      className="img-item-store"
      src="../assets/pizzas/pizza-1.png"
    />
  </div>
  <div
    className="flex-column"
  >
    <span
      className="title-item-store"
    >
      Pizza de Pollo
    </span>
  </div>
</div>
`;
